pycqed/analysis_v2/coherence_analysis.py

```diff
@@ -1,27 +1,302 @@
+        > T1, T2, Echo vs flux
+        > T1, T2, Echo vs frequency
+        > T1, T2, Echo vs flux sensitivity
+        > ratio Ramsey/Echo vs flux
+        > ratio Ramsey/Echo vs frequency
+        > ratio Ramsey/Echo vs flux sensitivity
+        > Dephasing rates Ramsey and Echo vs flux sensitivity
+        > Dac arc fit, use to assess if sensitivity calculation is correct.
+
+    If properties of resonator are provided (freq_resonator, Qc, chi_shift),
+    it also calculates the number of noise photons.
+    """
+
+    def __init__(self, coherence_table,
+                 t_start: str=None, t_stop: str=None, label='',
+                 options_dict: dict=None, auto: bool=True, close_figs=True,
+                 freq_resonator: float=None, Qc: float=None,
+                 chi_shift: float=None, savename: str= 'coherence_analysis',
+                 **kwargs):
+
+        super().__init__(t_start=t_start, t_stop=t_stop, label=label,
+                         options_dict=options_dict, close_figs=close_figs,
+                         **kwargs)
+        if t_stop is not None:
+            self.options_dict['save_figs'] = False
+        self._coherence_table = coherence_table
+        self._freq_resonator = freq_resonator
+        self._Qc = Qc
+        self._chi_shift = chi_shift
+
+        if auto:
+            self.run_analysis()
+
+    def extract_data(self):
+        """Put data from the coherence table in the raw data dict."""
+        self.raw_data_dict = OrderedDict()
+        rdd = self.raw_data_dict
+        rdd['dac'], rdd['freq'], rdd['T1'], rdd['Tramsey'],\
+            rdd['Techo'], rdd['exclusion_mask'] = self._coherence_table
+        # Externally loaded tables my be cast to ints.
+        rdd['exclusion_mask'] = rdd['exclusion_mask'].astype(bool)
+
+        # Resonator information
+        rdd['freq_resonator'] = self._freq_resonator
+        rdd['Qc'] = self._Qc
+        rdd['chi_shift'] = self._chi_shift
+
+        # Extra info for datasaving etc.
+        rdd['timestamps'] = [self.t_start, self.t_stop]
+        t_stop = self.t_stop.split('_')
+        folder = os.path.join(
+            a_tools.datadir, t_stop[0],
+            '{}_coherence_analysis'.format(t_stop[1]))
+        self.raw_data_dict['folder'] = [folder]
+        print(folder)
+
+    def process_data(self):
+        """
+        Process data.
+
+        Performs the following:
+            - fit dac-arc
+            - convert dac to flux
+            - calculate sensitivity δf/δΦ
+            - calculate dephasing rates Γ
+            - fit dephasing rates
+            - determine derived quantities
+        """
+        self.fit_res = OrderedDict()
+        self.proc_data_dict = deepcopy(self.raw_data_dict)
+        pdd = self.proc_data_dict
+
+        # Extract the dac arcs required for getting the sensitivities
+        # FIXME: add a proper guess function to make the fit more robust
+        self.fit_res['dac_arc'] = fit_frequencies(pdd['dac'], pdd['freq'],
```

Comment:
Fails to fit dac arc when I'm using dac that is already in flux quantum units. Needs better dac0_guess. Perhaps based on parabola fit?